src/plugins/graph/adornments/plotted-function/plotted-function-adornment-component.tsx

```diff
@@ -133,9 +106,15 @@ export const PlottedFunctionAdornmentComponent = observer(function PlottedFuncti
         const { domain: xDomain } = xAxis; // eslint-disable-line unused-imports/no-unused-vars
         const { domain: yDomain } = yAxis; // eslint-disable-line unused-imports/no-unused-vars
       }
+      // Trigger an autorun if the expression of y changes
+      // TODO Change this from hard coded "y"
+      if (sharedVariables) {
+        const y = sharedVariables.variables.find(variable => variable.name === "y");
+        y?.expression; // eslint-disable-line no-unused-expressions
```

Comment:
This doesn't need to be done in this PR, but if any of the inputs of y change (or their inputs, or their inputs...) then this should also trigger an update. 

I think it would work to read `y?.computedValueIncludingMessageAndError`. That will go "up" the tree of inputs to get the value. It will also be monitoring the expression.  So it should then trigger a refresh if any of those things change.

It is fine to put that change in a later PR.